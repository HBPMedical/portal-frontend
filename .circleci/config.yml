version: 2
jobs:
  build:
    docker:
      - image: circleci/buildpack-deps:xenial
    steps:
      - setup_remote_docker:
          docker_layer_caching: true
      - checkout
      - run:
          name: 'Build frontend'
          command: |
            docker build -t hbpmip/portal-frontend .

  test:
    docker:
      - image: circleci/buildpack-deps:xenial
    steps:
      - setup_remote_docker:
          docker_layer_caching: true
      - checkout
      - run:
          name: Setup Environment Variables
          command: |
            echo 'export HOST="CircleCI"' >> $BASH_ENV
      - run:
          name: 'Pull & install backend'
          command: |
            git clone https://github.com/HBPMedical/web-analytics-demo.git
            cp .circleci/server/* web-analytics-demo/
            cd web-analytics-demo
            ./ci.sh
      - run:
          name: 'Build frontend'
          command: |
            cd app/v3
            echo REACT_APP_BACKEND_URL=\"http://portalbackend:8080\" | tee -a .env
            docker build -t frontend:latest .
      - run:
          name: 'Test frontend'
          command: |
            docker run -e CI=true --network=webanalyticsstarter_default frontend:latest yarn test

workflows:
  version: 2
  build_and_test:
    jobs:
      - build
      - test
