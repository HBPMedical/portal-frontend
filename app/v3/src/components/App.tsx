import * as React from "react";
import "./App.css";

import { Experiment, Navigation } from "./";

class App extends React.Component {
  public render() {
    return (
      <div className="App">
        <Navigation />
        <Experiment />
      </div>
    );
  }
}

export default App;
