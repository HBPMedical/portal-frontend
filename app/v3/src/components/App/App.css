body {
  overflow-y: scroll !important;
}

@font-face {
  font-family: Open Sans Condensed;
  src: url(../../fonts/open-sans-condensed-v12-latin-700.eot);
  src: url(../../fonts/open-sans-condensed-v12-latin-700.eot#iefix)
      format("embedded-opentype"),
    url(../../fonts/open-sans-condensed-v12-latin-700.woff) format("woff"),
    url(../../fonts/open-sans-condensed-v12-latin-700.ttf) format("truetype"),
    url(../../fonts/open-sans-condensed-v12-latin-700.svg#OpenSans)
      format("svg");
  font-weight: 700;
  font-style: normal;
}

@font-face {
  font-family: themify;
  src: url(../../fonts/themify.eot);
  src: url(../../fonts/themify.eot#iefix) format("embedded-opentype"),
    url(../../fonts/themify.woff) format("woff"),
    url(../../fonts/themify.ttf) format("truetype"),
    url(../../fonts/themify.svg#OpenSans) format("svg");
  font-style: normal;
}

body {
  background: url(../../images/body-bg.jpg) top center no-repeat fixed #f5f5f5;
  background-size: 100% auto;
  overflow-x: hidden;
}

.App {
  background: url(../../images/body-bg.jpg) top center no-repeat fixed #f5f5f5;
  background-size: 100% auto;
  overflow-x: hidden;
}

.App section {
  margin-top: 116px;
}

body,
h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: "Open Sans", sans-serif;
  font-weight: 400;
  color: #616161;
}

a {
  color: #337ab7;
  text-decoration: none;
}
a:hover,
a:focus {
  color: #23527c;
  text-decoration: underline;
}
a:focus {
  outline: 5px auto -webkit-focus-ring-color;
  outline-offset: -2px;
}

h2,
h3 {
  margin-top: 20px;
  margin-bottom: 10px;
}
h5 {
  margin-top: 10px;
  margin-bottom: 10px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h5 {
  font-size: 14px;
}
p {
  margin: 0 0 10px;
}
ul {
  margin-top: 0;
  margin-bottom: 10px;
}
ul ul {
  margin-bottom: 0;
}

strong {
  font-weight: bold;
}
img {
  border: 0;
  vertical-align: middle;
}
svg:not(:root) {
  overflow: hidden;
}
button {
  margin: 0;
  font: inherit;
  color: inherit;
}
button {
  overflow: visible;
}
button {
  text-transform: none;
}
button {
  -webkit-appearance: button;
  cursor: pointer;
}
button::-moz-focus-inner {
  padding: 0;
  border: 0;
}

@media print {
  *,
  *:before,
  *:after {
    color: #000 !important;
    text-shadow: none !important;
    background: transparent !important;
    -webkit-box-shadow: none !important;
    box-shadow: none !important;
  }
  a,
  a:visited {
    text-decoration: underline;
  }
  a[href]:after {
    content: " (" attr(href) ")";
  }
  a[href^="#"]:after {
    content: "";
  }

  .navbar {
    display: none;
  }
  .btn > .caret {
    border-top-color: #000 !important;
  }
}

.panel {
  background-color: #fff;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.2);
  border: 0 !important;
  margin-bottom: 8px;
}

.panel-body {
  padding: 15px;
  font-size: 13px;
  position: relative;
}

.panel,
.panel .panel-body {
  border-radius: 4px;
  background-color: #fff;
  width: 100%;
}
.panel-title {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 16px;
  color: inherit;
}

.panel.panel-default {
  border-color: #fafafa;
  border-color: #ddd;
}
