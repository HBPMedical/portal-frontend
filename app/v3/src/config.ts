import * as dotenv from "dotenv";

dotenv.config();

// const trainingDatasetsValue = process.env.REACT_APP_TRAININGDATASETS;
// const trainingDatasets: string[] = trainingDatasetsValue
//   ? trainingDatasetsValue.split(",")
//   : [];
// const validationDatasetsValue = process.env.REACT_APP_VALIDATIONDATASETS;
// const validationDatasets: string[] = validationDatasetsValue
//   ? validationDatasetsValue.split(",")
//   : [];
const Cookie = process.env.REACT_APP_COOKIE;

const options: RequestInit =
  process.env.NODE_ENV === "production"
    ? {
        credentials: "same-origin"
      }
    : Cookie
      ? {
          headers: {
            Authorization: process.env.REACT_APP_AUTHORIZATION!,
            Cookie,
            "X-XSRF-TOKEN": Cookie.match(/XSRF-TOKEN=(.*)/)![1] || ""
          }
        }
      : {};

const baseUrl = `${process.env.REACT_APP_BACKEND_URL}`;

export default  { options, baseUrl };
