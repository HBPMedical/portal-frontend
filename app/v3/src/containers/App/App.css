.App {
  background: url(../../images/body-bg.jpg) top center no-repeat fixed #f5f5f5;
  background-size: 100% auto;
  overflow-x: hidden;
}

.App section {
  margin-top: 116px;
}

body,
h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: "Open Sans", sans-serif;
  font-weight: 400;
  color: #616161;
}

body {
  background: url(../../images/body-bg.jpg) top center no-repeat fixed #f5f5f5;
  background-size: 100% auto;
  overflow-x: hidden;
}
