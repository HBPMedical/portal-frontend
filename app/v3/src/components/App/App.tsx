// tslint:disable jsx-no-lambda

import * as React from 'react';
import { Route } from 'react-router-dom';
import { APICore, APIExperiment, APIMining, APIModel } from '../API';
import ExperimentCreate from '../Experiment/Create/Container';
import ExperimentResult from '../Experiment/Result/Container';
import ExperimentReview from '../Experiment/Review/Container';
import Explore from '../Explore/NativeBubble';
import Navigation from '../UI/Navigation';

import 'bootstrap/dist/css/bootstrap.css';
import './App.css';

interface IProps {
  appConfig: any;
  apiExperiment: APIExperiment;
  apiCore: APICore;
  apiModel: APIModel;
  apiMining: APIMining;
}

const App = ({
  appConfig,
  apiExperiment,
  apiCore,
  apiModel,
  apiMining
}: IProps) => (
  <div className='App'>
    <header>
      <Navigation
        apiExperiment={apiExperiment}
        apiModel={apiModel}
        appConfig={appConfig}
      />
    </header>
    <section>
      <Route path='/v3/explore' render={() => <Explore apiCore={apiCore} />} />
      <Route
        path='/v3/review'
        render={() => (
          <ExperimentReview
            apiMining={apiMining}
            apiModel={apiModel}
            apiCore={apiCore}
          />
        )}
      />
      <Route
        path='/v3/experiment/:slug/:uuid'
        render={() => (
          <ExperimentResult
            apiExperiment={apiExperiment}
            apiModel={apiModel}
            apiCore={apiCore}
          />
        )}
      />
      <Route
        exact={true}
        path='/v3/experiment/:slug'
        render={() => (
          <ExperimentCreate
            apiExperiment={apiExperiment}
            apiCore={apiCore}
            apiModel={apiModel}
            appConfig={appConfig}
          />
        )}
      />
    </section>
  </div>
);

export default App;
