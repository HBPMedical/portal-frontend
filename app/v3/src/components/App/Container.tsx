import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import request from 'request-promise-native';
import { Provider, Subscribe } from 'unstated';

// import UNSTATED from 'unstated-debug';
import { APICore, APIExperiment, APIMining, APIModel, webURL } from '../API'; // as interfaces
import { Pathology } from '../API/Core';
import config from '../API/RequestHeaders';
import App from '../App/App';

// UNSTATED.logStateChanges = process.env.NODE_ENV === "development";

interface State {
  appConfig: any;
}

class AppContainer extends React.Component<any, State> {
  public state: State = { appConfig: {} };
  private apiExperiment = new APIExperiment(config);
  private apiModel = new APIModel(config);
  private apiCore = new APICore(config);
  private apiMining = new APIMining(config);

  private intervalId: any; // FIXME: NodeJS.Timer | undefined;

  public async componentDidMount() {
    this.intervalId = setInterval(() => this.apiExperiment.all(), 10 * 1000);

    // Conf written by dockerize
    const json = await request.get(`${webURL}/scripts/app/config.json`);
    try {
      const appConfig = JSON.parse(json);
      this.setState({ appConfig });
    } catch (e) {
      const appConfig = {
        instanceName: 'MIP LOCAL',
        mode: 'local',
        theme: 'mip',
        version: 'alpha'
      };
      this.setState({ appConfig });
      console.log("Couldn't read config");
    }

    console.log(this.state.appConfig.mode);

    return await Promise.all([
      this.apiExperiment.all(),
      this.apiCore.datasets(),
      this.apiCore.setPathology(Pathology.DEG),
      this.apiCore.algorithms(this.state.appConfig.mode === 'local'),
      this.apiCore.stats(),
      this.apiModel.all()
    ]);
  }

  public componentWillUnmount() {
    clearInterval(this.intervalId);
  }

  public render() {
    return (
      <Router>
        <Provider
          inject={[
            this.apiExperiment,
            this.apiCore,
            this.apiModel,
            this.apiMining
          ]}>
          <Subscribe to={[APIExperiment, APICore, APIModel, APIMining]}>
            {(
              apiExperiment: APIExperiment,
              apiCore: APICore,
              apiModel: APIModel,
              apiMining: APIMining
            ) => {
              return (
                <App
                  appConfig={this.state.appConfig}
                  apiExperiment={apiExperiment}
                  apiCore={apiCore}
                  apiModel={apiModel}
                  apiMining={apiMining}
                />
              );
            }}
          </Subscribe>
        </Provider>
      </Router>
    );
  }
}

export default AppContainer;
