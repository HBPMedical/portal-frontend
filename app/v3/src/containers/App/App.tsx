// import "./Bootstrap-custom.css"
// tslint:disable:no-console

import "bootstrap/dist/css/bootstrap.css";
import * as React from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import { Provider, Subscribe } from "unstated";
import UNSTATED from "unstated-debug";
import {
  ExperimentContainer,
  ExperimentListContainer,
  ExploreContainer,
  MethodContainer,
  ModelContainer
} from "../";
import {
  Bubble,
  Experiment,
  Experiments,
  Graph,
  Navigation,
  RunExperiment,
  TreeMap
} from "../../components";
import config from "../../config";

import "./App.css";

UNSTATED.logStateChanges = process.env.NODE_ENV === "development";

class App extends React.Component {
  private experimentContainer = new ExperimentContainer(config);
  private experimentListContainer = new ExperimentListContainer(config);
  private modelContainer = new ModelContainer(config);
  private exploreContainer = new ExploreContainer(config);
  private methodContainer = new MethodContainer(config);

  private intervalId: NodeJS.Timer;

  public componentWillMount() {
    this.intervalId = setInterval(
      () => this.experimentListContainer.load(),
      10 * 1000
    );
  }

  public componentWillUnmount() {
    clearInterval(this.intervalId);
  }

  public render() {
    return (
      <Router>
        <Provider
          inject={[
            this.experimentContainer,
            this.experimentListContainer,
            this.exploreContainer,
            this.modelContainer,
            this.methodContainer
          ]}
        >
          <Subscribe
            to={[
              ExperimentContainer,
              ExperimentListContainer,
              ExploreContainer,
              ModelContainer,
              MethodContainer
            ]}
          >
            {(
              experimentContainer: ExperimentContainer,
              experimentListContainer: ExperimentListContainer,
              exploreContainer: ExploreContainer,
              modelContainer: ModelContainer,
              methodContainer: MethodContainer
            ) => (
              <div className="App">
                <header>
                  <Navigation
                    experimentContainer={experimentContainer}
                    experimentListContainer={experimentListContainer}
                    modelContainer={modelContainer}
                  />
                </header>
                <section>
                  <Route
                    path="/v3/explore"
                    // tslint:disable-next-line jsx-no-lambda
                    render={() => (
                      <Bubble exploreContainer={exploreContainer} />
                    )}
                  />
                  <Route
                    path="/v3/explore1"
                    // tslint:disable-next-line jsx-no-lambda
                    render={() => (
                      <TreeMap exploreContainer={exploreContainer} />
                    )}
                  />
                  <Route
                    path="/v3/explore2"
                    // tslint:disable-next-line jsx-no-lambda
                    render={() => <Graph exploreContainer={exploreContainer} />}
                  />
                  <Route
                    path="/v3/experiments"
                    // tslint:disable-next-line jsx-no-lambda
                    render={() => (
                      <Experiments
                        experimentListContainer={experimentListContainer}
                      />
                    )}
                  />
                  <Route
                    path="/v3/experiment/:slug/:uuid"
                    // tslint:disable-next-line jsx-no-lambda
                    render={() => (
                      <Experiment
                        experimentContainer={experimentContainer}
                        experimentListContainer={experimentListContainer}
                        modelContainer={modelContainer}
                      />
                    )}
                  />
                  <Route
                    exact={true}
                    path="/v3/experiment/:slug"
                    // tslint:disable-next-line jsx-no-lambda
                    render={() => (
                      <RunExperiment
                        experimentContainer={experimentContainer}
                        experimentListContainer={experimentListContainer}
                        exploreContainer={exploreContainer}
                        methodContainer={methodContainer}
                        modelContainer={modelContainer}
                      />
                    )}
                  />
                </section>
              </div>
            )}
          </Subscribe>
        </Provider>
      </Router>
    );
  }
}

export default App;
