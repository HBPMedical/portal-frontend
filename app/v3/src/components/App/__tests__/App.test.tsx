import {
  APICore,
  APIExperiment,
  APIMining,
  APIModel
} from "@app/components/API";
import config from "@app/components/API/RequestHeaders";
import App from "@app/components/App/App";
import AppContainer from "@app/components/App/Container";
import * as React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

jest.mock("request-promise-native");

describe("Test App component", () => {
  const apiExperiment = new APIExperiment(config);
  const apiModel = new APIModel(config);
  const apiCore = new APICore(config);
  const apiMining = new APIMining(config);
  const props = {
    appConfig: { appConfig: "MIP" },
    apiExperiment,
    apiCore,
    apiModel,
    apiMining
  };

  it("AppContainer dom renders correctly", () => {
    const wrapper = shallow(<AppContainer />);
    expect(wrapper).toBeDefined();
  });

  it("AppContainer snapshot renders correctly", () => {
    const tree = renderer.create(<AppContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("App dom renders correctly", () => {
    const wrapper = shallow(
        <App {...props} />
    );
    expect(wrapper).toBeDefined();
  });

  it("App snapshot renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <App {...props} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
