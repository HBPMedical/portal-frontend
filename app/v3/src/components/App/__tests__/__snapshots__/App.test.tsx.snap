// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component App snapshot renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <header
      className="navbar navbar-default navbar-fixed-top "
      id="topnav"
    >
      <div
        className="container-fluid"
      >
        <div
          className="logo-area"
        >
          <a
            className="navbar-brand"
            href="/home"
            onClick={[Function]}
          >
            Human Brain Project
          </a>
          <span
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="/data/mydata"
              onClick={[Function]}
              title="My data"
            >
              <span
                className="icon-bg"
              >
                <i
                  className="ti ti-menu"
                />
              </span>
            </a>
          </span>
          <span
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="/data/communitydata"
              onClick={[Function]}
              title="My community"
            >
              <span
                className="icon-bg"
              >
                <i
                  className="ti ti-world"
                />
              </span>
            </a>
          </span>
        </div>
        <div
          className="app-instance-name"
        >
          <h2>
            MIP
          </h2>
        </div>
        <ul
          className="nav navbar-nav toolbar pull-right"
        >
          <li
            className="toolbar-icon-bg hidden-xs"
          >
            <a
              href="/explore"
              onClick={[Function]}
              title="Epidemiological Exploration "
            >
              <span
                className="icon-bg"
              >
                <i>
                  EE
                </i>
              </span>
            </a>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs"
          >
            <a
              aria-current={null}
              className={undefined}
              href="/v3/review"
              onClick={[Function]}
              style={undefined}
              title="Interactive Analysis"
            >
              <span
                className="icon-bg"
              >
                <i>
                  IA
                </i>
              </span>
            </a>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs bsd"
          >
            <span
              title="Biological Signature of Diseases"
            >
              <div
                className="dropdown btn-group btn-group-default"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-default"
                  disabled={false}
                  id="experiment-dropdown"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  BSD
                </button>
                <ul
                  aria-labelledby="experiment-dropdown"
                  className="dropdown-menu"
                  role="menu"
                >
                  <div
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                    style={
                      Object {
                        "margin": "8px",
                      }
                    }
                  >
                    <p>
                      You have no running experiments.
                    </p>
                    <p>
                      You can start one by selecting a model and configuring an experiment on it.
                    </p>
                  </div>
                </ul>
              </div>
            </span>
          </li>
          <li
            className="uib-dropdown toolbar-icon-bg"
          >
            <a
              className="uib-dropdown-toggle username"
              data-position="right"
              href="#"
            >
              <img
                alt="anonymous"
                className="img-circle"
                src="default_user.png"
              />
            </a>
            <ul
              className="dropdown-menu userinfo arrow"
            >
              <li>
                <a
                  href="/profile"
                  ui-sref="profile"
                >
                  <i
                    className="ti ti-user"
                  />
                  <span>
                    Profile
                  </span>
                </a>
              </li>
              <li
                className="divider"
              />
              <li>
                <a
                  href=""
                  id="logout-link"
                  ng-click="logout()"
                >
                  <i
                    className="ti ti-shift-right"
                  />
                  <span>
                    Sign out
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="https://hbpmedical.github.io/documentation/HBP_SP8_UserGuide_latest.pdf"
              target="_help"
              title="Knowledge base"
            >
              <span
                className="icon-bg"
                style={
                  Object {
                    "fontWeight": 900,
                  }
                }
              >
                ?
              </span>
            </a>
          </li>
          <li
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="/hbpapps"
              title="App"
              ui-sref="hbpapps"
            >
              <span
                className="icon-bg"
              >
                <i
                  className="ti ti-layout-grid3-alt "
                  style={
                    Object {
                      "fontFamily": "themify",
                      "fontWeight": "normal",
                    }
                  }
                />
              </span>
            </a>
          </li>
        </ul>
      </div>
    </header>
  </header>
  <section />
</div>
`;

exports[`Test App component AppContainer snapshot renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <header
      className="navbar navbar-default navbar-fixed-top "
      id="topnav"
    >
      <div
        className="container-fluid"
      >
        <div
          className="logo-area"
        >
          <a
            className="navbar-brand"
            href="/home"
            onClick={[Function]}
          >
            Human Brain Project
          </a>
          <span
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="/data/mydata"
              onClick={[Function]}
              title="My data"
            >
              <span
                className="icon-bg"
              >
                <i
                  className="ti ti-menu"
                />
              </span>
            </a>
          </span>
          <span
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="/data/communitydata"
              onClick={[Function]}
              title="My community"
            >
              <span
                className="icon-bg"
              >
                <i
                  className="ti ti-world"
                />
              </span>
            </a>
          </span>
        </div>
        <div
          className="app-instance-name"
        >
          <h2>
            MIP
          </h2>
        </div>
        <ul
          className="nav navbar-nav toolbar pull-right"
        >
          <li
            className="toolbar-icon-bg hidden-xs"
          >
            <a
              href="/explore"
              onClick={[Function]}
              title="Epidemiological Exploration "
            >
              <span
                className="icon-bg"
              >
                <i>
                  EE
                </i>
              </span>
            </a>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs"
          >
            <a
              aria-current={null}
              className={undefined}
              href="/v3/review"
              onClick={[Function]}
              style={undefined}
              title="Interactive Analysis"
            >
              <span
                className="icon-bg"
              >
                <i>
                  IA
                </i>
              </span>
            </a>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs bsd"
          >
            <span
              title="Biological Signature of Diseases"
            >
              <div
                className="dropdown btn-group btn-group-default"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-default"
                  disabled={false}
                  id="experiment-dropdown"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  BSD
                </button>
                <ul
                  aria-labelledby="experiment-dropdown"
                  className="dropdown-menu"
                  role="menu"
                >
                  <div
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                    style={
                      Object {
                        "margin": "8px",
                      }
                    }
                  >
                    <p>
                      You have no running experiments.
                    </p>
                    <p>
                      You can start one by selecting a model and configuring an experiment on it.
                    </p>
                  </div>
                </ul>
              </div>
            </span>
          </li>
          <li
            className="uib-dropdown toolbar-icon-bg"
          >
            <a
              className="uib-dropdown-toggle username"
              data-position="right"
              href="#"
            >
              <img
                alt="anonymous"
                className="img-circle"
                src="default_user.png"
              />
            </a>
            <ul
              className="dropdown-menu userinfo arrow"
            >
              <li>
                <a
                  href="/profile"
                  ui-sref="profile"
                >
                  <i
                    className="ti ti-user"
                  />
                  <span>
                    Profile
                  </span>
                </a>
              </li>
              <li
                className="divider"
              />
              <li>
                <a
                  href=""
                  id="logout-link"
                  ng-click="logout()"
                >
                  <i
                    className="ti ti-shift-right"
                  />
                  <span>
                    Sign out
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="https://hbpmedical.github.io/documentation/HBP_SP8_UserGuide_latest.pdf"
              target="_help"
              title="Knowledge base"
            >
              <span
                className="icon-bg"
                style={
                  Object {
                    "fontWeight": 900,
                  }
                }
              >
                ?
              </span>
            </a>
          </li>
          <li
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="/hbpapps"
              title="App"
              ui-sref="hbpapps"
            >
              <span
                className="icon-bg"
              >
                <i
                  className="ti ti-layout-grid3-alt "
                  style={
                    Object {
                      "fontFamily": "themify",
                      "fontWeight": "normal",
                    }
                  }
                />
              </span>
            </a>
          </li>
        </ul>
      </div>
    </header>
  </header>
  <section />
</div>
`;
